### ISSUE TITLE
Persistent notifications aren't removed. Make Error while creating/copying file to warning dismissable

### ISSUE THREAD

**Issue by Findus23** [matomo-org/matomo#15575]
quite low priority, but I wanted to write it down just in case I stumble across it again in the future.
(and maybe the same is true for other error messages and it confuses user)

When developing Matomo doesn't have write permission to `plugins/`. This doesn't matter as it isn't needed for things apart from the marketplace.

When one tries to download something from the marketplace this error message appears rightfully:

![grafik](https://user-images.githubusercontent.com/6266037/74612027-b0212200-5101-11ea-865e-6396dec7bf23.png)

The issue is that there is no way to get rid of the notification. Pressing the x only closes it with JS, but as it is rendered by Twig it will reappear on every page even when it doesn't matter anymore. 

Adding a `static::$session->unsetAll();` to 
https://github.com/matomo-org/matomo/blob/b16a791aa3650d85af829156129c2bd44c7cb075/core/Notification/Manager.php#L191-L202

removes it from the session and gets finally rid of the message (ending the session in another way would also work).

**Comment by sgiehl** [comment]
maybe related to #15498

**Comment by Findus23** [comment]
@sgiehl That sounds like the same general issue (notifications in the session don't disappear)

**Comment by Findus23** [comment]
https://github.com/matomo-org/matomo/issues/12934 is also caused by the same issue.

**Comment by tsteur** [comment]
To fix this problem we want to fix the generic problem that a persistent notification only seems to disappear when logging out and logging in.

You can reproduce this by applying this change

```diff
diff --git a/plugins/CorePluginsAdmin/PluginInstaller.php b/plugins/CorePluginsAdmin/PluginInstaller.php
index 4619ecca9f..11f10a1ac6 100644
--- a/plugins/CorePluginsAdmin/PluginInstaller.php
+++ b/plugins/CorePluginsAdmin/PluginInstaller.php
@@ -52,6 +52,7 @@ class PluginInstaller
 
     public function installOrUpdatePluginFromMarketplace($pluginName)
     {
+        throw new \Exception('foo');
         $this->checkMarketplaceIsEnabled();
 
         $this->pluginName = $pluginName;
```

Then go to marketplace, and try to install a plugin. Then go back for example to the marketplace page or the Platform -> API page in the Matomo admin.

The notification will always appear again, even if you click on the "x". It seems it is supposed to call a method `CoreHome controller markNotificationAsRead action` when clicking on the "x" and then not appear anymore. However, it doesn't look like the notification angular component is actually sending that request maybe.

**Comment by tsteur** [comment]
![image](https://user-images.githubusercontent.com/273120/115322301-dfd22b00-a1d9-11eb-8bd5-999a7eff956e.png)
problem is there's no notificationId defined in the angularjs component. It's likely only set in the scope.

**Comment by tsteur** [comment]
@sgiehl is this one already done maybe?

**Comment by sgiehl** [comment]
Yes. That was actually already fixed with 4.3.0
